torch/onnx/_internal/fx/dynamo_exporter.py

```diff
@@ -28,41 +37,156 @@ def export(self) -> torch.onnx.ExportOutput:
 
         class GraphCaptureCompiler:
             def __init__(self):
-                self.captured_graph: Optional["torch.fx.GraphModule"] = None
+                self.captured_graph: Optional[torch.fx.GraphModule] = None
                 self.captured_graph_count = 0
 
-            def compile(self, graph_module: "torch.fx.GraphModule", _):
+            def compile(self, graph_module: torch.fx.GraphModule, _):
                 assert self.captured_graph_count == 0
                 self.captured_graph = graph_module
                 self.captured_graph_count += 1
                 return graph_module
 
         compiler = GraphCaptureCompiler()
         torch._dynamo.reset()
-        torch._dynamo.optimize(compiler.compile, nopython=True)(self.model)(*bound_args)
+        # TODO(titaiwang): Set `dynamic` according to `self.options.dynamic_shapes`
```

Comment:
Unrelated to this PR. I added this todo and I'm curious if it is accurate @titaiwangms ? It felt strange that export option `dynamic_shapes` does not affect dynamo `dynamic`/`tracing_mode`.

Reply:
I think this kind of circle back to the question that "Do we have to align dynamo and make_fx mode"? Based on my findings, we can pass dynamic test cases without setting this one dynamic, but I can try enable this.

Reply:
I think it certainly affects dynamo's behavior in capturing fx graphs with dynamic shapes. I suspect without alignment, we are only doing "dynamo static capture" + "refill symbolic shapes".

I feel turning this on in dynamo call won't be a trivial task. I can imagine it introducing a lot more graph breaks, which is resolvable with the new `constraints` api FYI https://github.com/pytorch/pytorch/pull/96699 if you are interested, and perhaps other patterns. Let's dig more and do this when we reach the step of refining `dynamic` apis for `dynamo_export`.

Reply:
Thanks for the info!

Reply:
I will be looking into this now.

Reply:
https://github.com/pytorch/pytorch/issues/98962